packages/client/src/store/leaderboard/LeaderboardActions.ts

```diff
@@ -0,0 +1,60 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { SCORE_FIELD_NAME } from 'src/constants/LeaderboardConsts';
+import { addError } from '../error/ErrorSlice';
+import { fetchApi } from '../utils';
+
+const defaultHeaders = {
+  'content-type': 'application/json',
+  mode: 'cors',
+};
+
+export interface IScore {
+  data: {
+    [SCORE_FIELD_NAME]: number;
+    user: string;
+  };
+  ratingFieldName: string;
+  teamName: string;
+}
+
+export interface ILeaderboardRequest {
+  ratingFieldName: string;
+  cursor: number;
+  limit: number;
+}
+
+export const recordScore = createAsyncThunk(
+  'leaderboard',
+  async (data: IScore, thunkApi) => {
+    try {
+      const res = await fetchApi('/leaderboard', {
```

Comment:
res unknoden, стоит передать тип возвращаемого значения в fetch 
```suggestion
      const res = await fetchApi<TypeForResponce>('/leaderboard', {
```